prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -395,22 +395,40 @@ class ActionHistory {
     }
 
     /**
-     * Automatically generate parent/child relationships based on what's in the reports* collections.
-     * For now, assume that every parent report played a hand in creating every child report.
-     * This is a lovely simplification, because it means that the functions don't have to
-     * worry about lineage tracking at all.
+     * Use the detailed item lineage to exactly/correctly generate the report parent/child relationships.
+     *
      */
     private fun generateReportLineages(actionId: Long) {
-        reportsIn.keys.forEach { parentId ->
-            reportsOut.keys.forEach { childId ->
-                reportLineages.add(ReportLineage(null, actionId, parentId, childId, null))
-            }
```

Comment:
Old code was written before we had item-lineages, so I just assumed every parent was the parent of every child.

Now, we have item-lineages, so I can just use that like a DNA test to extract exact maternity/paternity.